discord/threads.py

```diff
@@ -268,6 +268,17 @@ def is_news(self) -> bool:
         """
         return self._type is ChannelType.news_thread
 
+    def is_nsfw(self) -> bool:
+        """:class:`bool`: Whether the thread is NSFW or not.
+
+        An NSFW thread is a thread that has a parent that is an NSFW channel,
+        i.e. :meth:`.TextChannel.is_nsfw` is ``True``.
+        """
+        parent = self.parent
+        if parent is None:
+            raise ClientException('Parent channel not found')
+        return parent.is_nsfw()
```

Comment:
```suggestion
        parent = self.parent
        return parent is not None and parent.is_nsfw()
```

I figure this is better than throwing in a boolean method.